### PR TITLE
Add input-activate option to disable activation in input layer

### PR DESCRIPTION
**Pull request by jalencato**
*Issue #, if available:*
Add an option for GNN node input layer to disable activation layer and dropout

*Description of changes:*

For ogbn-arxiv, it enables the test accuracy raising from 64.46 to 64.68 for 2-layer RGCN.
57.12 to 64.12 for 3-layer RGCN

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: python/graphstorm/config/argument.py [hunk 1]
```diff
@@ -578,6 +579,19 @@ def model_encoder_type(self):
             f"Model encoder type should be in {BUILTIN_ENCODER}"
         return self._model_encoder_type
 
+    @property
+    def input_activate(self):
+        """ Whether include activation layer and dropout in the input layer
+        """
+        # pylint: disable=no-member
+        if hasattr(self, "_input_activate"):
+            assert self._input_activate in [True, False], \
```
Comment (classicsong) [inline python/graphstorm/config/argument.py:588]: Can we make it more flexible like self._input_activate.lower() in ["none", "relu"]. So we can add other activation if needed.

File: python/graphstorm/config/argument.py [hunk 2]
```diff
@@ -578,6 +579,19 @@ def model_encoder_type(self):
...
+                "Use input activate flag must be True or False"
+            return self._input_activate
+
+        # By default, use mini batch inference, which requires less memory
+        return True
```
Comment (classicsong) [inline python/graphstorm/config/argument.py:593]: return None by default.

Comment (jalencato) [reply python/graphstorm/config/argument.py:593]: I think here we should return relu as the default option.

File: docs/source/configuration/configuration-run.rst [hunk 3]
```diff
@@ -180,6 +180,10 @@ GraphStorm provides a set of parameters to control training hyper-parameters.
     - Yaml: ``num_ffn_layers_in_decoder: 1``
     - Argument: ``--num-ffn-layers-in-decoder 1``
     - Default value: ``0``
+- **input_activation**: Graphstorm provides this argument as an option to change the activation function in the input layer. Please note, it is only effective for 'relu' or 'none'.
```
Comment (classicsong) [inline docs/source/configuration/configuration-run.rst:183]:
```suggestion
- **input_activation**: Graphstorm provides this argument as an option to change the activation function in the input layer. Please note, it only accepts 'relu' and 'none'.
```

File: python/graphstorm/config/argument.py [hunk 4]
```diff
@@ -578,6 +579,19 @@ def model_encoder_type(self):
...
         return self._model_encoder_type
 
+    @property
+    def input_activate(self):
+        """ Design activation funtion type in the input layer
+        """
+        # pylint: disable=no-member
+        if hasattr(self, "_input_activate"):
+            assert self._input_activate in ["none", "relu"], \
+                "Use input activate flag for None and Relu"
```
Comment (classicsong) [inline python/graphstorm/config/argument.py:589]:
```suggestion
                "Use input activate flag 'none' for None and 'relu' for torch.nn.functional.relu"
```

File: python/graphstorm/config/argument.py [hunk 5]
```diff
@@ -578,6 +579,19 @@ def model_encoder_type(self):
...
+            return self._input_activate
```
Comment (classicsong) [inline python/graphstorm/config/argument.py:590]: Can we do something like:
```
if self._input_activate == "none":
    return None
elif self._input_activate == "relu":
    return F.relu
else:
    raise RuntimeError(""Use input activate flag 'none' for None and 'relu' for torch.nn.functional.relu")
```

File: docs/source/configuration/configuration-run.rst [hunk 6]
```diff
@@ -180,6 +180,10 @@ GraphStorm provides a set of parameters to control training hyper-parameters.
     - Yaml: ``num_ffn_layers_in_decoder: 1``
     - Argument: ``--num-ffn-layers-in-decoder 1``
     - Default value: ``0``
+- **input_activation**: Graphstorm provides this argument as an option to change the activation function in the input layer. Please note, it only accepts 'relu' and 'none'.
```
Comment (classicsong) [inline docs/source/configuration/configuration-run.rst:183]:
```suggestion
- **input_activate**: Graphstorm provides this argument as an option to change the activation function in the input layer. Please note, it only accepts 'relu' and 'none'.
```

File: docs/source/configuration/configuration-run.rst [hunk 7]
```diff
@@ -180,6 +180,10 @@ GraphStorm provides a set of parameters to control training hyper-parameters.
...
+    - Yaml: ``input_activation: relu``
```
Comment (classicsong) [inline docs/source/configuration/configuration-run.rst:184]:
```suggestion
    - Yaml: ``input_activate: relu``
```

File: docs/source/configuration/configuration-run.rst [hunk 8]
```diff
@@ -180,6 +180,10 @@ GraphStorm provides a set of parameters to control training hyper-parameters.
...
+    - Argument: ``--input-activation relu``
```
Comment (classicsong) [inline docs/source/configuration/configuration-run.rst:185]:
```suggestion
    - Argument: ``--input-activate relu``
```

File: tests/unit-tests/test_embed.py [hunk 9]
```diff
@@ -35,7 +37,7 @@
 from util import create_tokens
 
 # In this case, we only use the node features to generate node embeddings.
-def test_input_layer1():
+def test_input_layer1(input_activate=None):
```
Comment (classicsong) [inline tests/unit-tests/test_embed.py:40]: You can use
```
@pytest.mark.parametrize("input_activate", [None, F.relu])
```
To test both relu and none in CI

**Review by classicsong — APPROVED** [review]
